var/spack/repos/builtin/packages/cmake/package.py

```diff
@@ -294,21 +315,59 @@ def bootstrap_args(self):
 
         return args
 
+    def winbootcmake(self, spec):
+        from spack import fetch_strategy, stage
+        urls = {
+            '3': 'https://cmake.org/files/v3.21/cmake-3.21.2-windows-x86_64.zip',
+            '2': 'https://cmake.org/files/v2.8/cmake-2.8.4-win32-x86.zip'
+        }
+        if spec.satisfies('@3.0.2:'):
+            bootstrap_url = urls['3']
+        else:
+            bootstrap_url = urls['2']
+        remote = fetch_strategy.URLFetchStrategy(url=bootstrap_url)
+        bootstrap_stage_path = os.path.join(self.stage.path, "cmake-bootstraper")
+        with stage.Stage(remote, path=bootstrap_stage_path) as bootstrap_stage:
+            remote.stage = bootstrap_stage
+            remote.fetch()
+            remote.expand()
+            shutil.move(bootstrap_stage.source_path, self.stage.source_path)
+
+    def cmake_bootstrap(self):
+        exe_prefix = self.stage.source_path
+        relative_cmake_exe = os.path.join('spack-src', 'bin', 'cmake.exe')
+        return Executable(os.path.join(exe_prefix, relative_cmake_exe))
+
     def bootstrap(self, spec, prefix):
-        bootstrap = Executable('./bootstrap')
-        bootstrap(*self.bootstrap_args())
+        bootstrap_args = self.bootstrap_args()
+        if os.name == 'nt':
+            self.winbootcmake(spec)
+            bootstrap = self.cmake_bootstrap()
+            bootstrap_args.extend(['.'])
+        else:
+            bootstrap = Executable('./bootstrap')
+        bootstrap(*bootstrap_args)
 
     def build(self, spec, prefix):
```

Comment:
This build phase is not designed to specify arguments to cmake at what in `CMakePackages` would be the `cmake` phase but to ninja in the `build` phase. I think if anything it would be better if anything to rename the bootstrap phase above, but honestly I don't think that accurately describes what's going on because CMake is bootstrapping itself, as was indicated by the name prior to this pr.